File: campushub-frontend/src/services/facultyService.js
import api from "../api/axiosConfig";

export const getAllFaculties = async () => {
    const response = await api.get("/faculties");
    return response.data.data;
};

export const createFaculty = async (faculty) => {
    const response = await api.post("/faculties", faculty);
    return response.data.data;
};

export const updateFaculty = async (id, faculty) => {
    const response = await api.put(`/faculties/${id}`, faculty);
    return response.data.data;
};

export const deleteFaculty = async (id) => {
    const response = await api.delete(`/faculties/${id}`);
    return response.data.data;
};

export const getMyProfile = async () => {

    const response = await api.get("/faculties/me");

    return response.data.data;
};

export const getFacultyDashboard = async () => {

    const response = await api.get("/faculty/dashboard");

    return response.data.data;
};

File: src/main/java/com/preeti/campushub/service/FacultyService.java
package com.preeti.campushub.service;

import java.util.List;

import org.springframework.data.domain.Page;

import com.preeti.campushub.dto.faculty.FacultyRequest;
import com.preeti.campushub.dto.faculty.FacultyResponse;

public interface FacultyService {

    FacultyResponse createFaculty(FacultyRequest request);

    List<FacultyResponse> getAllFaculties();

    FacultyResponse getFacultyById(Long id);

    FacultyResponse updateFaculty(Long id, FacultyRequest request);

    void deleteFaculty(Long id);

    Page<FacultyResponse> getFaculty(
        int page,
        int size,
        String sortBy,
        String direction);

    List<FacultyResponse> searchFaculty(String keyword);

    List<FacultyResponse> filterFaculty(Long departmentId);

    FacultyResponse getMyProfile();
}